runtime/dart_isolate.cc

```diff
@@ -742,12 +743,32 @@ bool DartIsolate::RunFromLibrary(std::optional<std::string> library_name,
   auto entrypoint_handle = entrypoint.has_value() && !entrypoint.value().empty()
                                ? tonic::ToDart(entrypoint.value().c_str())
                                : tonic::ToDart("main");
+  auto entrypoint_args = tonic::ToDart(args);
   auto user_entrypoint_function =
       ::Dart_GetField(library_handle, entrypoint_handle);
 
-  auto entrypoint_args = tonic::ToDart(args);
+  // The Dart plugin registrant is a static with signature `void register()`
+  // within the class `_PluginRegistrant` generated by the Flutter tool.
+  //
+  // This method binds a plugin implementation to their platform
+  // interface based on the configuration of the app's pubpec.yaml, and the
+  // plugin's pubspec.yaml.
+  //
+  // Since this method may or may not be defined, check if the class is defined
+  // in the default library before calling the method.
+  Dart_Handle classRegistrant =
+      ::Dart_GetClass(library_handle, tonic::ToDart("_PluginRegistrant"));
+
+  if (!Dart_IsError(classRegistrant)) {
+    Dart_Handle result =
+        tonic::DartInvokeField(classRegistrant, "register", {});
+    if (tonic::LogIfError(result)) {
+      FML_LOG(ERROR) << "Could not invoke the main entrypoint.";
+    }
```

Comment:
This can just be

```suggestion
    tonic::LogIfError(tonic::DartInvokeField(class_registrant, "register", {});
```

The extra `FML_LOG` will just add more logging noise, unless we have some instructions to add here for users about what to do to fix this (rerun `flutter pub get`?)